packages/core/streams/openai-stream.ts

```diff
@@ -1,27 +1,7 @@
-import {
-  AIStream,
-  trimStartOfStreamHelper,
-  type AIStreamCallbacks
-} from './ai-stream'
+import type { AIStreamOptions } from './types'
+import { AIStream } from './ai-stream'
+import { openaiParsers } from './parsers'
 
-function parseOpenAIStream(): (data: string) => string | void {
-  const trimStartOfStream = trimStartOfStreamHelper()
-  return data => {
-    // TODO: Needs a type
-    const json = JSON.parse(data)
-
-    // this can be used for either chat or completion models
-    const text = trimStartOfStream(
-      json.choices[0]?.delta?.content ?? json.choices[0]?.text ?? ''
-    )
-
-    return text
-  }
-}
-
-export function OpenAIStream(
-  res: Response,
-  cb?: AIStreamCallbacks
-): ReadableStream {
-  return AIStream(res, parseOpenAIStream(), cb)
+export function OpenAIStream(res: Response, options: AIStreamOptions = {}) {
+  return AIStream(res, { parsers: openaiParsers, ...options })
```

Comment:
The `parsers` object contains everything `AIStream()` needs to parse chunks for each of the given `AIStreamMode`s.